core/src/main/java/io/aiven/klaw/service/SchemaRegstryControllerService.java

```diff
@@ -270,6 +276,82 @@ public ApiResponse execSchemaRequestsDecline(String avroSchemaId, String reasonF
     }
   }
 
+  public ApiResponse promoteSchema(SchemaPromotion schemaPromotion) throws Exception {
+    String userDetails = getUserName();
+
+    if (commonUtilsService.isNotAuthorizedUser(
+        getPrincipal(), PermissionType.REQUEST_CREATE_SCHEMAS)) {
+      return ApiResponse.builder().result(ApiResultStatus.NOT_AUTHORIZED.value).build();
+    }
+
+    Integer userTeamId = commonUtilsService.getTeamId(userDetails);
+    int tenantId = commonUtilsService.getTenantId(getUserName());
+    int teamId = commonUtilsService.getTeamId(userDetails);
+    SchemaRequestModel schemaRequest =
+        buildSchemaRequestFromPromotionRequest(schemaPromotion, teamId, tenantId);
+
+    Optional<Env> optionalEnv = getSchemaEnvFromId(schemaPromotion.getSourceEnvironment());
+
+    if (!optionalEnv.isPresent()) {
+      return ApiResponse.builder()
+          .status(HttpStatus.BAD_REQUEST)
+          .result("Unable to find or access the source Schema Registry")
+          .build();
+    }
+    Env schemaEnv = optionalEnv.get();
+    SortedMap<Integer, Map<String, Object>> schemaObjects =
+        getSchemasFromTopicName(schemaPromotion.getTopicName(), tenantId, schemaEnv);
+    log.info(
+        "getSchemaVersion {}, schemaObjects keySet {}",
+        schemaPromotion.getSchemaVersion(),
+        schemaObjects.keySet());
+    Map<String, Object> schemaObject =
+        schemaObjects.get(Integer.valueOf(schemaPromotion.getSchemaVersion()));
+    // Pretty Print the Json String so that it can be seen clearly in the UI.
+    schemaRequest.setSchemafull(prettyPrintUglyJsonString((String) schemaObject.get("schema")));
```

Comment:
am not sure if we need prettyprint here. may be we have another pretty print when viewing schema requests

Reply:
So that pretty print is needed or else the UI shows the schema as 1 long string in the UI and it's really hard to review.
I think it lost its formatting when saved to the database and as thats where I was retrieving the schema from it was needed to make it human readable.